src/adapter/src/catalog/storage.rs

```diff
@@ -1386,17 +1386,17 @@ impl<'a> Transaction<'a> {
     pub(crate) fn update_item(
         &mut self,
         id: GlobalId,
-        item_name: &str,
-        item: &SerializedCatalogItem,
+        entry: &catalog::CatalogEntry,
     ) -> Result<(), Error> {
         let n = self.items.update(|k, v| {
             if k.gid == id {
+                let definition = Catalog::serialize_item(entry.item());
                 Some(ItemValue {
                     schema_id: v.schema_id,
-                    name: item_name.to_string(),
-                    definition: item.clone(),
-                    owner_id: v.owner_id,
-                    privileges: v.clone().privileges,
+                    name: entry.name().item.clone(),
+                    definition,
+                    owner_id: *entry.owner_id(),
+                    privileges: entry.privileges().all_values_owned().collect(),
```

Comment:
IIUC the fix here is the provided `CatalogEntry` has the updated privileges, but the provided value `v` does not. So privileges would exist in the in-memory catalog, but were never actually getting persisted?

Reply:
Yeah, `v` is the current item in the stash, so we were always updating it to the same exact value.